src/execution/operator/persistent/buffered_csv_reader.cpp

```diff
@@ -540,18 +530,28 @@ vector<LogicalType> BufferedCSVReader::SniffCSV(vector<LogicalType> requested_ty
 		// set all sql_types to VARCHAR so we can do datatype detection based on VARCHAR values
 		sql_types.clear();
 		sql_types.assign(options.num_cols, LogicalType::VARCHAR);
-		InitParseChunk(sql_types.size());
 
-		// detect types in first chunk
-		JumpToBeginning(options.skip_rows, false);
+		// jump to beginning and skip potential header
+		JumpToBeginning(options.skip_rows, true);
+		DataChunk header_row;
+		header_row.Initialize(sql_types);
+		parse_chunk.Copy(header_row);
+
+		// init parse chunk and read csv with info candidate
+		InitParseChunk(sql_types.size());
 		ParseCSV(ParserMode::SNIFFING_DATATYPES);
-		for (idx_t row = 0; row < parse_chunk.size(); row++) {
+		for (int64_t row = -1; row < static_cast<int64_t>(parse_chunk.size()); row++) {
```

Comment:
Perhaps change the iteration here back to being 0-based, and use row==0 for the header row and `row - 1` for the remaining rows? Perhaps could have something like this:
```cpp
for (idx_t ridx = 0; ridx <= parse_chunk.size(); ridx++) {
   bool is_header = ridx == 0;
   idx_t row = ridx - 1;
   ...
}
```